src/graph/GoExecutor.cpp

```diff
@@ -574,24 +570,22 @@ void GoExecutor::stepOut() {
     };
     auto error = [this] (auto &&e) {
         LOG(ERROR) << "Exception when handle out-bounds/in-bounds: " << e.what();
-        doError(Status::Error("Exeception when handle out-bounds/in-bounds: %s.",
+        doError(Status::Error("Exception when handle out-bounds/in-bounds: %s.",
                     e.what().c_str()));
     };
     std::move(future).via(runner).thenValue(cb).thenError(error);
 }
 
 
 void GoExecutor::onStepOutResponse(RpcResponse &&rpcResp) {
+    joinResp(std::move(rpcResp));
+
     if (isFinalStep()) {
-        maybeFinishExecution(std::move(rpcResp));
+        maybeFinishExecution();
         return;
     } else {
-        auto status = getDstIdsFromResp(rpcResp);
-        if (!status.ok()) {
-            doError(std::move(status).status());
-            return;
-        }
-        starts_ = std::move(status).value();
+        auto dsts = getDstIdsFromResps(records_.end() - 1, records_.end());
```

Comment:
modify `records_.end() - 1` to `records_.rbegin()`

Reply:
I don't think it's better clear.